### PR TITLE
Porting applicable fixes from SignalR server 2.2.1 

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Messaging.Tests/SubscriptionFacts.cs [hunk 1]
```diff
@@ -40,14 +40,14 @@ public void CancelledTaskShouldThrowTaskCancelledSync()
         }
 
         [Fact]
-        public void CancelledTaskShouldThrowTaskCancelledAsync()
+        public async Task CancelledTaskShouldThrowTaskCancelledAsync()
         {
             Func<MessageResult, object, Task<bool>> callback = async (result, state) =>
             {
-                await Task.Delay(500);
+                await Task.Delay(500); // .ConfigureAwait(false);
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Messaging.Tests/SubscriptionFacts.cs]: Remove dead code

**Comment by davidfowl** [comment]
One minor comment but LGTM  🚢 🇮🇹

File: src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs [hunk 2]
```diff
@@ -117,6 +117,16 @@ private IEnumerable<Type> GetTypesSafe(Assembly a)
                                      ex.GetType().Name,
                                      ex.Message);
 
+                if (ex.LoaderExceptions != null)
+                {
+                    _logger.LogWarning("Loader exceptions messages: ");
+
+                    foreach (var exception in ex.LoaderExceptions)
+                    {
+                        _logger.LogWarning("{0}\r\n", exception);
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs]: Environment.NewLine?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs]: Should I change it also above (ln. 110) then?

File: test/Microsoft.AspNetCore.SignalR.Messaging.Tests/SubscriptionFacts.cs (hunk 1 above)
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Messaging.Tests/SubscriptionFacts.cs]: It's already fixed in https://github.com/aspnet/SignalR-Server/pull/203. I just need to rebase.

File: src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs (hunk 2 above)
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs]: 108-110
Probably :D

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/Lookup/ReflectedHubDescriptorProvider.cs]: Will do.

**Pull request merged**